lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -482,14 +516,146 @@ namespace Intl
         error = UErrorCode::U_ZERO_ERROR;
 
         written = uloc_toLanguageTag(defaultLocaleId, bcp47, ULOC_FULLNAME_CAPACITY, true, &error);
-        ICU_ASSERT_RETURN(error, written > 0 && written < cchLangtag, 0);
+        ICU_ASSERT_RETURN(error, written  > 0 && ((size_t) written) < cchLangtag, 0);
 
         return utf8::DecodeUnitsIntoAndNullTerminateNoAdvance(
             langtag,
             reinterpret_cast<LPCUTF8>(bcp47),
             reinterpret_cast<LPCUTF8>(bcp47 + written)
         );
     }
+
+    bool CollatorGetCollation(_In_z_ const char *langtag, _Out_ char *collation, _In_ size_t cchCollation)
+    {
+        Assert(langtag != nullptr && collation != nullptr && cchCollation >= 0 && cchCollation < ULOC_FULLNAME_CAPACITY);
```

Comment:
>Assert [](start = 8, length = 6)

What happens if this assert fails?

Reply:
Since we control the callsite, I don't think they're ever expected to fail any more than any other assert in the codebase -- If any of these ever happen, it means something is distressingly wrong elsewhere.